google/cloud/storage/benchmarks/storage_latency_benchmark.cc

```diff
@@ -419,11 +428,11 @@ Options ParseArgs(int argc, char* argv[]) {
        }},
       {"--enable-connection-pool", "enable the client library connection pool",
        [&options](std::string const& val) {
-         options.enable_connection_pool = gcs_bm::ParseBoolean(val, true);
+         options.enable_connection_pool = *gcs_bm::ParseBoolean(val, true);
```

Comment:
Dereferencing a `StatusOr` that contains an error is technically UB. AFAICT, there's no guarantee that the parse will succeed here.

One alternative would be to have ParseBoolean() return an `optional<bool>`, and not take a default argument. That is
`optional<bool> ParseBoolean(std::string const&)`.

That way, a caller could invoke it like this:

```cc
options.enable_connection_pool = gcs_bm::ParseBoolean(val).value_or(true);
```

What do you think?

Reply:
It will require a little more work but I think the best solution here is to propagate the `Status`/`StatusOr`, regardless of how we to choose to represent the default argument.

Reply:
This is what I got so far and it has a few problems:

```C++
       [&options](std::string const& val) -> google::cloud::Status {
         auto status = gcs_bm::ParseBoolean(val, true);
         if (status && *status) {
           options.enable_xml_api = *status;
         }
         return status.status();
       }},
```

We're using `StatusOr<bool>` here, which you don't recommend obviously. Furthermore, there is a lot of boilerplate here.

I will go for your suggestion instead.